internal/dot/graph.go

```diff
@@ -105,6 +109,12 @@ type FailedNodes struct {
 	// TransitiveFailures is the list of nodes that failed to build due to
 	// missing/failed dependencies.
 	TransitiveFailures []*Result
+
+	//FailedCtors stores a map of failed Ctor Ids  that is populated as the graph is traversed for errors.
+	FailedCtors map[CtorID]struct{}
+
+	// FailedGroups stores a map of failed groupKeys that is populated as the graph is traversed for errors.
+	FailedGroups map[groupKey]struct{}
```

Comment:
Do both these fields need to be private. Red flag here is that groupKey is private so FailedGroups can never be used outside this package anyway.

Reply:
I believe in this comment you are implying that both fields should be private? I believe it should be an am changing that now.